### DISCUSSION TITLE
Libvirt and golanglint-ci issues

### DISCUSSION THREAD

**Discussion by benschlueter** [golangci/golangci-lint#3462] [Q&A] [answered]
Hello,
not sure if it is an issue, but I am unable to run golanglint-ci in github actions (locally everything works fine)

```
Run golangci/golangci-lint-action@v3.2.0
prepare environment
run golangci-lint
  Running [/home/runner/golangci-lint-1.50.1-linux-amd64/golangci-lint run --out-format=github-actions] in [] ...
  level=warning msg="[runner] Can't run linter goanalysis_metalinter: buildir: failed to load package libvirt: could not load export data: no export data for \"libvirt.org/go/libvirt\""
  level=error msg="Running error: 1 error occurred:\n\t* can't run linter goanalysis_metalinter: buildir: failed to load package libvirt: could not load export data: no export data for \"libvirt.org/go/libvirt\"\n\n"
  
  Error: golangci-lint exit with code 3
  Ran golangci-lint in 136976ms
```
I don't understand the error, and it might be possible that I've configured something wrongly.

[yaml-file](https://github.com/benschlueter/delegatio/blob/main/.github/workflows/lint.yml)

**Comment by bombsimon** [comment, marked as answer]
The Go package `libvirt.org/go/libvirt` provides CGo binding from the OS native Libvirt API. This means that you need to install [`libvirt`](https://libvirt.org/compiling.html) for your code to compile and be able to lint. Judging by your provided yaml file you're not doing this so you need to add steps for this. I'm not familiar with how to install this dependency but in its simplest form, for an OS like Ubuntu all you would have to do is add the following steps before linting:

```yaml
- name: Install [some dependency]
  run: sudo apt-get install -y [dependency]
```

**Reply by benschlueter** [reply]
Thanks! Appreciated.

**Reply by benschlueter** [reply]
FYI: the corresponding package for `Ubuntu 22.02` is `libvirt-dev`
